java/code/src/com/suse/manager/utils/PackageUtils.java

```diff
@@ -41,6 +43,40 @@ public static boolean isTypeDeb(Package pkg) {
         return "deb".equals(getArchTypeLabel(pkg));
     }
 
+    /**
+     * Get the channel architecture as a universal arch string, practically
+     * stripping away any arch type suffixes (e.g. "-deb")
+     * @param arch the package arch
+     * @return an package architecture string
+     */
+    public static String getUniversalArchString(PackageArch arch) {
+        return StringUtils.substringBeforeLast(arch.getLabel(), "-deb");
+    }
+
+    /**
+     * Return an EVR string representation in the format "[epoch:]version-release",
+     * stripping away any dummy release strings (e.g. "-X")
+     *
+     * @param evr the package evr
+     * @return string representation of epoch, version and release
+     */
+    public static String getUniversalEvrString(PackageEvr evr) {
```

Comment:
Is there a particular reason not to put this method in `PackageEvr`? In particular, do we really need both this and the original `toString`?

Reply:
Not really, but `toString` wouldn't be a good idea since we do some conditional substring operation, whereas `toString` automatically means it would be output as-is no matter what.

Moving both methods to respective classes with names `toUniversal...String`